js/ascendex.js

```diff
@@ -467,53 +483,96 @@ module.exports = class ascendex extends Exchange {
+            if (!perpetual) {
                 symbol = base + '/' + quote;
+            } else {
+                const lotSizeFilter = this.safeValue (market, 'lotSizeFilter');
+                minQty = this.safeNumber (lotSizeFilter, 'minQty');
+                maxQty = this.safeNumber (lotSizeFilter, 'maxQty');
+                const priceFilter = this.safeValue (market, 'priceFilter');
+                minPrice = this.safeNumber (priceFilter, 'minPrice');
+                maxPrice = this.safeNumber (priceFilter, 'maxPrice');
+                const str = this.safeString (market, 'underlying');
+                const substrings = str.split ('/');
+                baseId = substrings[0];
+                quoteId = substrings[1];
+                base = this.safeCurrencyCode (baseId);
+                quote = this.safeCurrencyCode (quoteId);
+                symbol = base + '/' + quote + ':' + settle;
             }
             const fee = this.safeNumber (market, 'commissionReserveRate');
             result.push ({
                 'id': id,
                 'symbol': symbol,
                 'base': base,
                 'quote': quote,
+                'settle': settle,
                 'baseId': baseId,
                 'quoteId': quoteId,
-                'info': market,
+                'settleId': settleId,
                 'type': type,
                 'spot': spot,
                 'margin': margin,
-                'future': future,
-                'active': active,
-                'precision': precision,
+                'swap': perpetual,
+                'futures': false,
+                'option': false,
+                'derivative': derivativeContract,
+                'contract': derivativeContract,
+                'linear': linear,
+                'inverse': undefined,
                 'taker': fee,
                 'maker': fee,
+                'contractSize': undefined,
+                'active': active,
+                'expiry': undefined,
+                'fees': undefined,
+                'info': market,
+                'precision': precision,
                 'limits': {
+                    'leverage': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
                     'amount': {
-                        'min': this.safeNumber (market, 'minQty'),
-                        'max': this.safeNumber (market, 'maxQty'),
+                        'min': minQty,
+                        'max': maxQty,
                     },
                     'price': {
-                        'min': this.safeNumber (market, 'tickSize'),
-                        'max': undefined,
+                        'min': minPrice,
+                        'max': maxPrice,
                     },
                     'cost': {
-                        'min': this.safeNumber (market, 'minNotional'),
-                        'max': this.safeNumber (market, 'maxNotional'),
```

Comment:
don't remove this

Reply:
When I left it in it made the entire limits section return [object Object]

Reply:
I can try to replace it with a work around

Reply:
> When I left it in it made the entire limits section return [object Object]

That's only on the command line, I think you need to use `--details` to get the actual object